src/sidebar/util/annotation-sharing.js

```diff
@@ -1,47 +1,78 @@
 /**
+ * @typedef {import('../../types/api').Annotation} Annotation
  * @typedef {import('../../types/config').HostConfig} HostConfig
  */
 
 import serviceConfig from '../service-config';
 
 /**
- * Is the sharing of annotations enabled? Check for any defined `serviceConfig`,
- * but default to `true` if none found.
+ * Retrieve an appropriate sharing link for this annotation.
  *
- * @param {HostConfig} settings
- * @return {boolean}
+ * If the annotation is on a shareable document (i.e. its document is
+ * web-accessible), prefer the `incontext` (bouncer) link, but fallback to the
+ * `html` (single-annotation `h` web view) link if needed.
+ *
+ * If the annotation is not on a shareable document, don't use the `incontext`
+ * link as that won't work; only use the single-annotation-view `html` link.
+ *
+ * Note that `html` links are not provided by the service for third-party
+ * annotations.
+ *
+ * @param {Annotation} annotation
+ * @return {string|undefined}
  */
-export function sharingEnabled(settings) {
-  const serviceConfig_ = serviceConfig(settings);
-  if (serviceConfig_ === null) {
-    return true;
+export function annotationSharingLink(annotation) {
+  if (isShareableURI(annotation.uri)) {
+    return annotation.links?.incontext ?? annotation.links?.html;
+  } else {
+    return annotation.links?.html;
   }
-  if (typeof serviceConfig_.enableShareLinks !== 'boolean') {
-    return true;
+}
+
+/**
+ * Generate a URI for sharing: a bouncer link built to share annotations in
+ * a specific group (groupID) on a specific document (documentURI). If the
+ * `documentURI` provided is not a web-accessible URL, no link is generated.
+ *
+ * @param {string} documentURI
+ * @param {string} groupID
+ * @return {string}
+ */
+export function pageSharingLink(documentURI, groupID) {
+  if (!isShareableURI(documentURI)) {
+    return '';
```

Comment:
This works, although I'd be inclined to return `null` instead and make it more obvious that callers need to handle the case where there is no link.

Reply:
When I was writing this, I was half-thinking the same thing, so, yeah, sounds good.